### PR TITLE
feat(toast): add additional options to global config (closes #3169)

### LINKED ISSUE #3169: Setting custom <b-toast> defaults
### Is your feature request related to a problem? Please describe...

When using toats on demand (using the ```this.$bvToast.toast()```), I always use my same custom behavior that is different from the defaults (such as position, title , duration...) but currently there's no way to set them initially and I have to re-set them in every call.

### Describe the solution you'd like

I would like to be able to make a call in my application loading code in order to set my deaults.

### Describe alternatives you've considered

If this is actually possible, I haven't found it in the documentation page.

### Additional context

This toast are soooo cool! :)

### PR DESCRIPTION
**Pull request by tmorehouse**
### Describe the PR

Adds additional values to global config for `b-toast`, `b-toaster` (and `this.$bvToast.toast()` as it uses the previous two).

Closes #3169

### PR checklist

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [ ] Feature
- [x] Enhancement
- [ ] ARIA accessibility
- [ ] Documentation update
- [ ] Other (please describe)

**Does this PR introduce a breaking change?** (check one)

- [x] No
- [ ] Yes (please describe)

**The PR fulfills these requirements:**

- [x] It's submitted to the `dev` branch, **not** the `master` branch
- [x] When resolving a specific issue, it's referenced in the PR's title (i.e. `[...] (fixes #xxx[,#xxx])`, where "xxx" is the issue number)
- [x] It should address only one issue or feature. If adding multiple features or fixing a bug and adding a new feature, break them into separate PRs if at all possible.
- [x] The title should follow the [**Conventional Commits**](https://www.conventionalcommits.org/) naming convention (i.e. `fix(alert): not alerting during SSR render`, `docs(badge): update pill examples, fix typos`, `chore: fix typo in README`, etc). **This is very important, as the `CHANGELOG` is generated from these messages.**

**If new features/enhancement/fixes are added or changed:**

- [ ] Includes documentation updates (including updating the component's `package.json` for slot and event changes)
- [ ] New/updated tests are included and passing (if required)
- [x] Existing test suites are passing
- [x] The changes have not impacted the functionality of other components or directives
- [ ] ARIA Accessibility has been taken into consideration (Does it affect screen reader users or keyboard only users? Clickable items should be in the tab index, etc.)

**If adding a new feature, or changing the functionality of an existing feature, the PR's
description above includes:**

- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

### CODE REVIEW

File: src/components/toast/toast.js [hunk 1]
```diff
@@ -32,7 +32,7 @@ export const props = {
   },
   variant: {
     type: String,
-    default: null
+    default: () => getComponentConfig(NAME, 'variant') || null
```
Comment (jackmu95) [inline src/components/toast/toast.js]: I think we should omit the fallbacks, since we don't do that for the other components.

Comment (tmorehouse) [reply src/components/toast/toast.js]: True...

File: src/components/toast/toast.js [hunk 2]
```diff
@@ -30,14 +30,14 @@ export const props = {
     type: String,
     default: null
   },
-  variant: {
-    type: String,
-    default: null
-  },
   toaster: {
     type: String,
     default: () => getComponentConfig(NAME, 'toaster') || 'b-toaster-top-right'
```
Comment (jackmu95) [inline src/components/toast/toast.js]: Please remove that too.

Comment (tmorehouse) [reply src/components/toast/toast.js]: Although a toaster name is required... and users could set it to null or an empty string by mistake.

Comment (jackmu95) [reply src/components/toast/toast.js]: But it adds additional bundle size, which de did with the config stuff already enough.

Comment (tmorehouse) [reply src/components/toast/toast.js]: I could force it to a string

Comment (jackmu95) [reply src/components/toast/toast.js]: This will prevent the prop type warning for invalid value in DEV-mode.
Not sure about that change.

Comment (tmorehouse) [reply src/components/toast/toast.js]: I'm not sure the prop type check runs on the `default` value or not.

Comment (tmorehouse) [reply src/components/toast/toast.js]: But it will prevent PortalTarget from crashing ;-)

Comment (tmorehouse) [reply src/components/toast/toast.js]: Unless we add a computed prop that stringifies the name, and we pass that computed prop toPortal and PortalTarget

**Pull request merged**